sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/config/CosmosConfig.java

```diff
@@ -17,32 +17,39 @@ public class CosmosConfig {
 
     private final boolean queryMetricsEnabled;
 
+    private final int maxDegreeOfParallelism;
+
     /**
      * Initialization
      *
      * @param responseDiagnosticsProcessor must not be {@literal null}
      * @param queryMetricsEnabled must not be {@literal null}
+     * @param maxDegreeOfParallelism must not be {@literal null}
      */
-    @ConstructorProperties({"responseDiagnosticsProcessor", "queryMetricsEnabled"})
+    @ConstructorProperties({"responseDiagnosticsProcessor", "queryMetricsEnabled", "maxDegreeOfParallelism"})
     public CosmosConfig(ResponseDiagnosticsProcessor responseDiagnosticsProcessor,
-                        boolean queryMetricsEnabled) {
-        this(responseDiagnosticsProcessor, null, queryMetricsEnabled);
+                        boolean queryMetricsEnabled,
+                        int maxDegreeOfParallelism) {
+        this(responseDiagnosticsProcessor, null, queryMetricsEnabled, maxDegreeOfParallelism);
     }
 
     /**
      * Initialization
      *
      * @param responseDiagnosticsProcessor must not be {@literal null}
-     * @param databaseThroughputConfig may be @{literal null}
+     * @param databaseThroughputConfig may be {@literal null}
      * @param queryMetricsEnabled must not be {@literal null}
+     * @param maxDegreeOfParallelism must not be {@literal null}
      */
-    @ConstructorProperties({"responseDiagnosticsProcessor", "databaseThroughputConfig", "queryMetricsEnabled"})
+    @ConstructorProperties({"responseDiagnosticsProcessor", "databaseThroughputConfig", "queryMetricsEnabled", "maxDegreeOfParallelism"})
```

Comment:
Same here, we should not change the existing constructor.

Reply:
Done.